File: receipt-analyzer-service/src/main/java/pl/madzierski/daniel/app/receipt/scan_resolver/impl/UserResolver.java
package pl.madzierski.daniel.app.receipt.scan_resolver.impl;

import org.springframework.stereotype.Service;
import pl.madzierski.daniel.app.receipt.revision.model.ReceiptRevisionResolveData;
import pl.madzierski.daniel.app.receipt.scan_resolver.ReceiptResolverStrategy;
import pl.madzierski.daniel.app.receipt.scan_resolver.ReceiptResolverStrategyType;

import java.util.List;

@Service
public class UserResolver implements ReceiptResolverStrategy {

    @Override
    public ReceiptResolverStrategyType strategy() {
        return ReceiptResolverStrategyType.USER;
    }

    @Override
    public ReceiptRevisionResolveData execute(List<String> filePath) {
        return ReceiptRevisionResolveData.builder().build();
    }
}

File: receipt-analyzer-service/src/main/kotlin/pl/madzierski/daniel/app/receipt/scan_resolver/impl/UserResolver.kt
package pl.madzierski.daniel.app.receipt.scan_resolver.impl

import org.springframework.stereotype.Service
import pl.madzierski.daniel.app.receipt.revision.model.ReceiptRevisionResolveData
import pl.madzierski.daniel.app.receipt.scan_resolver.ReceiptResolverStrategyType
import pl.madzierski.daniel.app.receipt.scan_resolver.ReceiptResolverStrategyTypeStrategy

@Service
class UserResolver: ReceiptResolverStrategyTypeStrategy {
    override fun strategy(): ReceiptResolverStrategyType = ReceiptResolverStrategyType.USER

    override fun execute(filePath: String): ReceiptRevisionResolveData {
        return ReceiptRevisionResolveData(null, null, null, null)
    }
}